tests/DcmInterpolationTest.cpp

```diff
@@ -0,0 +1,423 @@
+  ok = ok && planner.setPlannerPeriod(conf.dT);
+  ok = ok && planner.setMinimumAngleForNewSteps(conf.minAngleVariation);
+  ok = ok && planner.setMinimumStepLength(conf.minStepLength);
+  ok = ok && planner.setNominalWidth(conf.nominalWidth);
+  planner.addTerminalStep(true);
+  planner.startWithLeft(conf.swingLeft);
+
+  return ok;
+}
+
+/**
+ * Configure the interpolator
+ * return true in case of success, false otherwise
+ */
+bool configureInterpolator(FeetInterpolator& interpolator, const Configuration &conf)
+{
+  bool ok = true;
+  
+  ok = ok && interpolator.setSwitchOverSwingRatio(conf.switchOverSwingRatio);
+  ok = ok && interpolator.setTerminalHalfSwitchTime(conf.lastStepSwitchTime);
+  ok = ok && interpolator.setStepHeight(conf.stepHeight);
+  ok = ok && interpolator.setPauseConditions(conf.maxStepDuration, conf.nominalDuration);
+  ok = ok && interpolator.setCoMHeightSettings(conf.comHeight, conf.comHeightDelta);
+
+  // REMOVE ME
+  iDynTree::Vector2 leftZMPstance, rightZMPstance;
+  leftZMPstance(0) = conf.lStancePositionX;
+  leftZMPstance(1) = conf.lStancePositionY;
+  rightZMPstance(0) = conf.rStancePositionX;
+  rightZMPstance(1) = conf.rStancePositionY;
+  ok = ok && interpolator.setStanceZMPDelta(leftZMPstance, rightZMPstance);
+
+  // REMOVE ME
+  iDynTree::Vector2 leftZMPswitch, rightZMPswitch;
+  leftZMPswitch(0) = conf.lSwitchInitPositionX;
+  leftZMPswitch(1) = conf.lSwitchInitPositionY;
+  rightZMPswitch(0) = conf.rSwitchInitPositionX;
+  rightZMPswitch(1) = conf.rSwitchInitPositionY;
+  ok = ok && interpolator.setInitialSwitchZMPDelta(leftZMPswitch, rightZMPswitch);
+
+  ok = ok && interpolator.setFootLandingVelocity(conf.stepLandingVelocity);
+  ok = ok && interpolator.setFootApexTime(conf.footApexTime);
+  
+  return ok;
+}
+
+/**
+ * Save all the trajectories in files
+ */
+void printTrajectories(const FeetInterpolator& interpolator, size_t& newMergePoint, size_t mergePoint, DcmInitialState& boundaryConditionAtMergePoint,
+		       const std::string& pLFileName, const std::string& pRFileName,
+		       const std::string& heightFileName, const std::string& heightAccFileName,
+		       const std::string& dcmPosFileName, const std::string& dcmVelFileName,
+		       const std::string& mergePointFileName)
+{
+  // instantiate ofstream
+  std::ofstream posLeftStream, posRightStream;
+  std::ofstream heightStream, heightAccelerationStream;
+  std::ofstream dcmPosStream, dcmVelStream;
+  std::ofstream mergePointsStream;
+  
+  // open files
+  posLeftStream.open(pLFileName.c_str());
+  posRightStream.open(pRFileName.c_str());
+    
+  heightStream.open(heightFileName.c_str());
+  heightAccelerationStream.open(heightAccFileName.c_str());
+
+  dcmPosStream.open(dcmPosFileName.c_str());
+  dcmVelStream.open(dcmVelFileName.c_str());
+
+  mergePointsStream.open(mergePointFileName.c_str());
+
+  // print feet trajectories
+  static std::vector<iDynTree::Transform> lFootTrajectory, rFootTrajectory;  
```

Comment:
I know this is just a test, but why do you use a static object here?

Reply:
I can answer here. The test is conceived to run the generation of trajectory more than once. The new trajectories don't override completely the old, so it is necessary to keep also the old trajectories in memory to test the continuity. Since we are in a function and those objects are accessed only there, I thought of using ``static`` objects.